2011-07-25  Aurelien Jarno  <aurelien@aurel32.net>

	* res_send.c(__libc_res_nsend): Don't initialize maps for IPv6 
	nameservers.

diff --git a/resolv/res_send.c b/resolv/res_send.c
index a001c1e..43ff082 100644
--- a/resolv/res_send.c
+++ b/resolv/res_send.c
@@ -422,9 +422,9 @@ __libc_res_nsend(res_state statp, const u_char *buf, int buflen,
 			}
 		}
 		n = statp->nscount;
-		if (statp->nscount > EXT(statp).nscount)
+		if (statp->nscount > EXT(statp).nscount + EXT(statp).nscount6)
 			for (n = EXT(statp).nscount, ns = 0;
-			     n < statp->nscount; n++) {
+			     n < statp->nscount - EXT(statp).nscount6; n++) {
 				while (ns < MAXNS
 				       && EXT(statp).nsmap[ns] != MAXNS)
 					ns++;
